src/api/layers/layers-external.api.js

```diff
@@ -38,34 +62,53 @@ export function parseWmsCapabilities(content, originUrl) {
     try {
         return new WMSCapabilitiesParser(content, originUrl)
     } catch (error) {
-        log.error(`Failed to parse WMS Get Capabilities`, error)
-        return null
+        throw new Error(`Failed to parse WMS Get Capabilities: ${error}`)
     }
 }
 
 /**
- * Read and parse WMTS GetCapabilities
+ * Sets the WMTS url parameters
  *
- * @param {string} baseUrl Base URL for the WMTS server
- * @returns {Promise<WMTSCapabilitiesParser | null>} WMTS Capabilities
+ * @param {URL} url Url to set
+ * @param {string} language Language to use
+ * @returns {URL} Url with wmts parameters
  */
-export async function readWmtsCapabilities(baseUrl) {
-    const url = new URL(baseUrl)
+export function setWmtsUrlParameters(url, language) {
```

Comment:
same as above, I think it would be better to state somehow that this is for getCapabilites request only (so that when the auto-complete of IDE provides it, we can know what this is for, and not use it for some other WM(T)S use cases)

Reply:
Done